go-controller/pkg/libovsdbops/switch.go

```diff
@@ -348,21 +349,22 @@ type logicalSwitchPortPredicate func(*nbdb.LogicalSwitchPort) bool
 func DeleteLogicalSwitchPortsWithPredicateOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, sw *nbdb.LogicalSwitch, p logicalSwitchPortPredicate) ([]libovsdb.Operation, error) {
 	sw, err := GetLogicalSwitch(nbClient, sw)
 	if err != nil {
-		return nil, fmt.Errorf("error getting switch %+v: %v", sw, err)
+		if errors.Is(err, libovsdbclient.ErrNotFound) {
+			return ops, nil
+		} else {
+			return nil, fmt.Errorf("error getting switch %+v: %v", sw, err)
```

Comment:
Please use `%w` instead of `%v`  to format err. It allows us to look for it if the caller cares about it.
```
return nil, fmt.Errorf("error getting switch %+v: %w", sw, err)
```